NitroxClient/Communication/Packets/Processors/CyclopsChangeEngineModeProcessor.cs

```diff
@@ -0,0 +1,56 @@
+﻿using NitroxClient.Communication.Packets.Processors.Abstract;
+using NitroxModel.Helper;
+using NitroxModel.Helper.GameLogic;
+using NitroxModel.Helper.Unity;
+using NitroxModel.Packets;
+using UnityEngine;
+
+namespace NitroxClient.Communication.Packets.Processors
+{
+    public class CyclopsChangeEngineModeProcessor : ClientPacketProcessor<CyclopsChangeEngineMode>
+    {
+        private PacketSender packetSender;
+
+        public CyclopsChangeEngineModeProcessor(PacketSender packetSender)
+        {
+            this.packetSender = packetSender;
+        }
+
+        public override void Process(CyclopsChangeEngineMode motorPacket)
+        {
+            GameObject cyclops = GuidHelper.RequireObjectFrom(motorPacket.Guid);
+            CyclopsHelmHUDManager hud = cyclops.RequireComponentInChildren<CyclopsHelmHUDManager>();
+            CyclopsMotorMode motorMode = cyclops.RequireComponentInChildren<CyclopsMotorMode>();
+            CyclopsMotorModeButton[] motorModeButtons = cyclops.RequireComponentsInChildren<CyclopsMotorModeButton>();
+
+            if (motorPacket.Mode != motorMode.cyclopsMotorMode)
+            {
+                foreach (CyclopsMotorModeButton motorModeButton in motorModeButtons)
+                {
+                    if (Player.main.currentSub == (SubRoot)motorModeButton.ReflectionGet("subRoot"))
+                    {
+                        motorModeButton.SetCyclopsMotorMode((CyclopsMotorMode.CyclopsMotorModes)motorPacket.Mode);
+                    }
+                    else
+                    {
+                        if (motorPacket.Mode == motorModeButton.motorModeIndex)
+                        {
+                            motorMode.cyclopsMotorMode = motorPacket.Mode;
```

Comment:
This code looks very similar to `CyclopsMotorMode.ChangeCyclopsMotorMode`. To be more future proof, it might be a better idea to call that function.

Even better is to reuse the onclick logic. Simply replace the entire foreach loop with:
```C#
motorMode.BroadcastMessage("SetCyclopsMotorMode", this.motorModeIndex, SendMessageOptions.RequireReceiver);
```
All the buttons will now update their own icon/state accordingly, and the selected button will in turn broadcast `ChangeCyclopsMotorMode` to the `CyclopsMotorMode` object again, which updates all the variables like you do here.

As an example of future-proof-ness, you are already missing `.SendMessage("RecalculateNoiseValues", null, SendMessageOptions.RequireReceiver);`.